src/file-events/cpp/win_fsnotifier.cpp

```diff
@@ -341,6 +341,7 @@ void Server::runLoop() {
 struct Command {
     Server* server;
     function<bool()> function;
+    mutex executionMutex;
```

Comment:
We can still share the `mutex`, let's use a `mutex*` here.

Reply:
I've thought about that, but this actually makes more sense to me. It's all pretty theoretical, because there will be very few calls through this anyway. But this way we don't need to hold up multiple threads trying to start/stop watching things just to schedule their command.

What's probably more important is that the `conditional_variable` is next to the `mutex` it belongs to.

Reply:
If it works, it is fine with me.